internal/registry/obj_selector.go

```diff
@@ -15,24 +15,29 @@ type ObjFilter struct {
 }
 
 type ObjSelector struct {
-	ctx     context.Context
-	objChan chan *ObjectInfo
-	boltDB  *bbolt.DB
-	filter  *ObjFilter
+	ctx       context.Context
+	objChan   chan *ObjectInfo
+	boltDB    *bbolt.DB
+	filter    *ObjFilter
+	cacheSize int
 }
 
-// objectSelectCache is a maximum number of the selected objects to be
+// objectSelectCache is a default number of the selected objects to be
```

Comment:
"maximum" was not added by accident, it is an expected situation when there are not enough objects to fill the cache

Reply:
Rephrased a bit